server/src/main/java/org/elasticsearch/indices/recovery/RecoveryState.java

```diff
@@ -671,7 +692,7 @@ public String toString() {
         }
     }
 
-    public static class Index extends Timer implements ToXContentFragment, Streamable {
+    public static class Index extends Timer implements ToXContentFragment, Streamable, Writeable {
```

Comment:
I think we can remove `Streamable` from `Index` then remove `readFrom` entirely.